File: 8.Best-Practice-e-trucchi-avanzati/objc/CM_Animation/CM_Animation WatchKit Extension/InterfaceController.m
//
//  Copyright © 2016 WhiteTiger "sgama la rete". All rights reserved.
//

#import "InterfaceController.h"

@interface InterfaceController()

@property (nonatomic, weak) IBOutlet WKInterfaceButton *btnMela;

@property (nonatomic, assign) CGSize maxSize;
@property (nonatomic, assign) CGSize minSize;

@end


@implementation InterfaceController

- (void)awakeWithContext:(id)context {
    [super awakeWithContext:context];
    
    self.maxSize = (CGSize){ 100.0, 100.0 };
    self.minSize = (CGSize){ 50.0, 50.0 };
}

- (void)willActivate {
    [super willActivate];
}

- (void)didDeactivate {
    [super didDeactivate];
}

#pragma mark - IBAction Methods

- (IBAction)actionZoom {
    [self animateWithDuration:0.5
                   animations:^{
                       [self.btnMela setWidth:self.maxSize.width];
                       [self.btnMela setHeight:self.maxSize.height];
                   }];
    
    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.5 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
        [self animateWithDuration:0.5
                       animations:^{
                           [self.btnMela setWidth:self.minSize.width];
                           [self.btnMela setHeight:self.minSize.height];
                       }];
    });
}

@end

File: 8.Best-Practice-e-trucchi-avanzati/swift/CM_Animation/CM_Animation WatchKit Extension/InterfaceController.swift
//
//  Copyright © 2015 WhiteTiger "sgama la rete". All rights reserved.
//

import WatchKit
import Foundation

class InterfaceController: WKInterfaceController {
    
    let maxSize: (width: CGFloat, height: CGFloat) = (100.0, 100.0)
    let minSize: (width: CGFloat, height: CGFloat) = (50.0, 50.0)
    
    @IBOutlet var btnMegaman: WKInterfaceButton!
    
    override func awakeWithContext(context: AnyObject?) {
        super.awakeWithContext(context)
    }
    
    override func willActivate() {
        super.willActivate()
    }
    
    override func didDeactivate() {
        super.didDeactivate()
    }
    
    // MARK: IBAction Methods
    
    @IBAction func actionZoom() {
        self.animateWithDuration(0.5) { () -> Void in
            self.btnMegaman.setWidth(self.maxSize.width)
            self.btnMegaman.setHeight(self.maxSize.height);
            
            dispatch_after(dispatch_time(DISPATCH_TIME_NOW, Int64(0.5 * Double(NSEC_PER_SEC))), dispatch_get_main_queue(), { () -> Void in
                self.animateWithDuration(0.5) { () -> Void in
                    self.btnMegaman.setWidth(self.minSize.width)
                    self.btnMegaman.setHeight(self.minSize.height);
                }
            })
        }
    }
}
